specifyweb/specify/migrations/0002_geo.py

```diff
@@ -0,0 +1,139 @@
+            )
+            cot.save()
+
+        # Iteratively update CollectionObjects' collectionobjecttype for each discipline in a bulk operation
+        for discipline in Discipline.objects.all():
+            discipline_name = discipline.name
+            cot = CollectionObjectType.objects.get(name=discipline_name)
+            (Collectionobject.objects
+                .filter(collection__discipline=discipline)
+                .update(cotype=cot))
+
+    operations = [
+        migrations.CreateModel(
+            name='CollectionObjectType',
+            fields=[
+                ('id', models.AutoField(db_column='CollectionObjectTypeID', primary_key=True, serialize=False)),
+                ('name', models.CharField(db_column='Name', max_length=255)),
+                ('isloanable', models.BooleanField(blank=True, db_column='IsLoanable', null=True)),
+                ('isdefault', models.BooleanField(blank=True, db_column='IsDefault', null=False, default=False)),
+                ('version', models.IntegerField(blank=True, db_column='Version', default=0, null=True)),
+                ('timestampcreated', models.DateTimeField(db_column='TimestampCreated', default=django.utils.timezone.now)),
+                ('timestampmodified', models.DateTimeField(blank=True, db_column='TimestampModified', default=django.utils.timezone.now, null=True)),
+                ('text1', models.TextField(blank=True, db_column='Text1', null=True)),
+                ('text2', models.TextField(blank=True, db_column='Text2', null=True)),
+                ('text3', models.TextField(blank=True, db_column='Text3', null=True)),
+                ('collection', models.ForeignKey(db_column='CollectionID', on_delete=specifyweb.specify.models.protect_with_blockers, related_name='collectionobjecttypes', to='specify.collection')),
+                ('createdbyagent', models.ForeignKey(db_column='CreatedByAgentID', null=True, on_delete=specifyweb.specify.models.protect_with_blockers, related_name='+', to='specify.agent')),
+                ('modifiedbyagent', models.ForeignKey(db_column='ModifiedByAgentID', null=True, on_delete=specifyweb.specify.models.protect_with_blockers, related_name='+', to='specify.agent')),
+                ('taxontreedef', models.ForeignKey(db_column='TaxonTreeDefID', on_delete=specifyweb.specify.models.protect_with_blockers, related_name='collectionobjecttypes', to='specify.taxontreedef')),
+            ],
+            options={
+                'db_table': 'collectionobjecttype',
+                'ordering': (),
+                'unique_together': (('collection', 'isdefault'),),
+            },
+        ),
+        migrations.AddField(
+            model_name='collectionobject',
+            name='cotype',
+            field=models.ForeignKey(db_column='COTypeID', null=True, on_delete=specifyweb.specify.models.protect_with_blockers, related_name='collectionobjects', to='specify.collectionobjecttype'),
```

Comment:
Do we want COs in a COG to be blocked from deletion until removed from the COG? How should this be handled?

Reply:
ya, that's a good point, it should probably be changed to `on_delete=models.DO_NOTHING`

Reply:
DO_NOTHING? 
Wouldn't that preserve the `COTypeID` on the CollectionObject when the COType is deleted? 
And thus the COTypeID would reference  CollectionObjectType which doesn't exist. 

I think SET_NULL is probably more appropriate 😅 

Unless some custom behavior for this is going to be implemented elsewhere in the API: (like how WorkBench deletion is handled)

Reply:
lol, ya, sorry, I meant SET_NULL, thanks for the catch!